### PR TITLE
refactor: remove and reduce usage of SCSS variables 

### PR DESCRIPTION
**Pull request by xitij2000**
This PR replaces usages of scss variables with closest equivalent classes. The aim is to be visually identical. 

There are places where this is not possible, and I hope this PR can start a discussion about what to do in those cases.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @xitij2000!  I've created [OSPR-6659](None/browse/OSPR-6659) to keep track of it in JIRA, where we prioritize reviews.  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by natabene** [comment]
@xitij2000 Thank you for your contribution. Is this ready for our review?

**Comment by xitij2000** [comment]
> @xitij2000 Thank you for your contribution. Is this ready for our review?

Not yet, I think I might break this into two PRs.

File: src/course-home/outline-tab/widgets/FlagButton.scss [hunk 1]
```diff
@@ -3,19 +3,18 @@
 @import "~@edx/brand/paragon/overrides";
 
 .flag-button {
-  background-color: $white;
-  border: 1px solid $light-400;
-  border-radius: .2rem;
+  box-shadow: 0 0 0 2px color-mix(in srgb, white 25%, var(--light));
   box-shadow: 0 0 0 2px $light-400;
```
Comment (tecoholic) [inline src/course-home/outline-tab/widgets/FlagButton.scss:6]:
```suggestion
```
Looks like a duplicate style definition.

File: src/course-home/outline-tab/widgets/FlagButton.scss [hunk 2]
```diff
@@ -3,19 +3,18 @@
...
 
   &:hover {
+    border: 1px solid color-mix(in srgb, white 50%, var(--primary));
     border: 1px solid $primary-300;
```
Comment (tecoholic) [inline src/course-home/outline-tab/widgets/FlagButton.scss:9]:
```suggestion
```
Another duplicate definition

File: src/course-home/outline-tab/widgets/FlagButton.scss [hunk 3]
```diff
@@ -3,19 +3,18 @@
...
-    box-shadow: 0 0 0 2px $white;
+    box-shadow: 0 0 0 2px var(--white);
   }
 }
 
 .flag-button-selected {
-  border: 1px solid $primary-300;
+  box-shadow: 0 0 0 2px color-mix(in srgb, white 50%, var(--primary));
   box-shadow: 0 0 0 2px $primary-300;
```
Comment (tecoholic) [inline src/course-home/outline-tab/widgets/FlagButton.scss:15]:
```suggestion
```
Another duplicate definition.

**Review by tecoholic — APPROVED** [review]
@xitij2000 :+1: with the small fixes for the duplicates

- [x] I tested this: Built the front-end, visited various pages to check if the elements are styled appropriately  & verified that most are as expected.
- [x] I read through the code
- [x] I checked for accessibility issues
- [ ] Includes documentation - NA

File: src/course-home/outline-tab/widgets/FlagButton.scss (hunk 1 above)
Comment (xitij2000) [reply src/course-home/outline-tab/widgets/FlagButton.scss:6]: @tecoholic Yes, in this case, the issue is that color-mix isn't supported currently, so this is more of a forward-looking change, but a better solution would be to just create css variables for colour mixes like --light-400 etc. 

I think it's worth discussing, and adding this to paragon.

**Comment by natabene** [comment]
Just checking if it still not ready.

**Comment by xitij2000** [comment]
> Just checking if it still not ready.

I'm currently preparing an ADR that could impact this PR, so will come back to it once that work is ready.

**Comment by natabene** [comment]
@xitij2000 Just wanted to check if this is still not ready.

**Comment by itsjeyd** [comment]
Hey @xitij2000, do you have any updates about this PR? Is it ready for review now, or perhaps no longer relevant?

**Comment by itsjeyd** [comment]
@xitij2000 Friendly ping about the status of this PR :) Is it ready for review or should we close it?

**Comment by xitij2000** [comment]
> @xitij2000 Friendly ping about the status of this PR :) Is it ready for review or should we close it?

Sorry for the delay. I've been recovering from illness recently and am trying to juggle quite a bit. I'll update these PRs next sprint.

**Comment by itsjeyd** [comment]
@xitij2000 Sounds good! Thanks for the update.

**Comment by xitij2000** [comment]
> @xitij2000 Just wanted to check if this is still not ready.

This is ready now.

**Comment by ProductRyan** [comment]
Adding to our internal JIRA for tracking:
https://2u-internal.atlassian.net/browse/AU-941

**Comment by itsjeyd** [comment]
@ProductRyan Just checking in to see if you had any updates on when we could expect a review for this PR?

@xitij2000 In the meantime could you address the merge conflicts that came up here?

File: src/generic/upsell-bullets/UpsellBullets.jsx [hunk 4]
```diff
@@ -46,7 +48,7 @@ export function UnlockGradedBullet() {
     </span>
   );
   return (
-    <li className="upsell-bullet">
```
Comment (leangseu-edx) [inline src/generic/upsell-bullets/UpsellBullets.jsx:49]: I would prefer this to remain `upsell-bullet`. Although the styling is trivial, we might use it as selector.

File: src/generic/upgrade-notification/UpgradeNotification.jsx [hunk 5]
```diff
@@ -15,7 +15,7 @@ import {
 
 function UpsellNoFBECardContent() {
   return (
-    <ul className="fa-ul upgrade-notification-ul pt-0">
```
Comment (leangseu-edx) [inline src/generic/upgrade-notification/UpgradeNotification.jsx:18]: Even if you remove `upgrade-notification-ul` style completely, I think these name should remain.

**Review by leangseu-edx — CHANGES_REQUESTED** [review]
I would ask you to preserve some of the className. Especially for those with class name of `upsell` and `upgrade`. They might be use for analytics.

**Comment by itsjeyd** [comment]
@xitij2000 Friendly ping on this. Looks like it's pretty close to being ready for merge :)

**Comment by itsjeyd** [comment]
Hey @xitij2000, when you get a minute could you please post an update on when you expect to get back to this PR?

**Comment by xitij2000** [comment]
> Hey @xitij2000, when you get a minute could you please post an update on when you expect to get back to this PR?

Since the creation of this PR, there have been new developments on the paragon side, with design tokens that will provide a better path for this kind of work. So I can close this PR for now and re-open when this can be rebased on that work.

**Comment by openedx-webhooks** [comment]
@xitij2000 Even though your pull request wasn’t merged, please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/frontend-app-learning&entry.867055334=https://github.com/openedx/frontend-app-learning/pull/942&entry.1484655318=https://github.com/xitij2000&entry.752974735=2022-05-05+12:57&entry.1917517419=2023-02-24+04:09&entry.2133058324=No) so we can improve your experience in the future.

**Pull request closed without merging**